Correctly print array and object SQL parameters in case of an exception.

RatkoR opened the pull request:
Laravel tries to show SQL that caused an exception. While doing it
it also prints all SQL parameters. If there are arrays or objects
in parameter it throws an "Object of class stdClass
could not be converted to string" exception.

This commit json_encodes all non scalar parameters so that they can
be printed in exception.

Commit:
Correctly print array and object SQL parameters in case of an exception.

Laravel tries to show SQL that caused an exception. While doing it
it also prints all SQL parameters. If there are arrays or objects
in parameter it throws an "Object of class stdClass
could not be converted to string" exception.

This commit json_encodes all non scalar parameters so that they can
be printed in exception.

cweiske commented on src/RatkoR/Crate/Connection.php:
```diff
@@ -195,4 +197,43 @@ public function query()
+        // If an exception occurs when attempting to run a query, we'll format the error
+        // message to include the bindings with SQL, which will make this exception a
+        // lot more helpful to the developer instead of just the database's errors.
+        catch (Exception $e) {
```
RatkoR\Crate does not have its own exception class.

RatkoR replied:
Fixed. Seems it was too late last night... Thx!

The test is also fixed and now properly catches RatkoR\Crate\QueryException. Any other issue you might see?

cweiske commented on tests/DataTests/DataTest.php:
```diff
@@ -298,4 +298,17 @@ public function it_tests_prepared_statements_without_params()
+
+    /**
+     * @test
+     * @expectedException Crate\PDO\Exception\PDOException
```
shouldn't this be a queryexception?

cweiske commented:
it would be nice if you had a travis-ci.org config file here so tests would run automatically.

Commit:
Adds missing use for Exception class.

Commit:
Fix: use stringified arrays and objects.

String version of arrays and objects were created but not used.
Changed method name to prepareBindings() from valuesToStrings()
because it doesn't cast to string all bindings, just arrays and
objects.

Commit:
Fixes exception test.

What we return is really an object of RatkoR\Crate\QueryException class
PDOException was thrown because there was an error in exception path.

RatkoR commented:
I haven't really worked with travis, as we don't use it here at our company. Which crate server would travis use when running tests?

cweiske commented:
The test script would actually need to start it itself. I [once did that for ldap](http://cweiske.de/tagebuch/ldap-server-travis.htm) but that was quite some task. [crate itself uses travis](https://crate.io/a/how-we-test-crate-data/), so maybe one can copy ideas from their .travis.yml file.

But yes, it won't be easy.

cweiske commented:
:+1: works fine here. Tested it by unsetting `$body[stmt]` in crate-pdo's `Http/Client.php`

cweiske commented:
(please squash the commits when merging so that it's only a single commit)

cweiske commented:
Thanks for acting so quickly.
